Contributing.md

```diff
@@ -0,0 +1,137 @@
+don't forget to do so when you clone a new sandbox!
+
+## commitlint
+
+The team has also agreed to use commitlint-style commit messages. Install and enable
+[commitlint](https://github.com/conventional-changelog/commitlint) to have your commits
+validated systematically.
+
+Commitlint commits look like this:
+
+    type(optional-scope): subject (i.e., short description)
+
+    optional body, which is free form
+
+    optional footer
+
+Valid types: (these are the default, which we're using as is for now)
+ - build: commits for the build system
+ - chore: maintain the repo, not the code itself
+ - ci: commits for the continuous integration system
+ - docs: adding and changing documentation
+ - feat: adding a new feature
+ - fix: fixing something
+ - improvement: improve something
+ - perf: improving performance
+ - refactor: refactor code
+ - revert: undo a previous change
+ - style: working only on code or documentation style
+ - test: commits for testing code
+
+Valid scopes: the scope is optional and usually refers to which module is being changed.
+ - TBD - for now not validated, should be just one word ideally
+
+Valid subject: short, free form, what the commit is about in less than 50 or 60 characters
+(not strictly enforced, but it's best to keep it short)
+
+Optional body: this is where you put all the verbose details you want about the commit, or
+nothing at all if the subject already says it all. Must be separated by a blank line from
+the subject. Explain what the changes are, why you're doing them, etc, as necessary.
+
+Optional footer: separated from the body (or subject if body is empty) by a blank line,
+lists reference (e.g.: "Closes #12" "Ref #24") or warns of breaking changes (e.g.,
+"BREAKING CHANGE: explanation").
+
+These rules are inspired by these commit formatting guides:
+ - [Conventional Commits](https://www.conventionalcommits.org/)
+ - [Bluejava commit guide](https://github.com/bluejava/git-commit-guide)
+ - [develar's commit message format](https://gist.github.com/develar/273e2eb938792cf5f86451fbac2bcd51)
+ - [AngularJS Git Commit Message Conventions](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y).
+
+### Enabling commitlint
+
+We run commitlint on each commit message that you write by enabling the commit-msg hook in
+Git. It is run via [husky](https://www.npmjs.com/package/husky), which is a JS Git hook
+manager, and you need Node to run it.
+
+If you don't already use Node, this is a bit more work to install that the pre-commit
+hooks above, but please take a moment to do this:
+
+- If you don't already use Node or nvm, install nvm in your ~/.nvm folder:
+```sh
+wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash`
+```
+This will add a few lines to your `.bashrc` file, which you'll need to execute now,
+possibly by starting a new shell.
+
+- Install Node, preferably version 12.18.3
```

Comment:
I feel like the docs here can just use `nvm install node`. I only said to use 12.18.3 because it's the current version I'm working on. But, since we use basically no JavaScript except for the commitlint package, I think we're fairly safe just recommending the latest stable version. Plus it means they don't have to do `nvm use 12.18.3`, so it's one less step

Reply:
what is nvm? do you mean npm?

Reply:
@dhdaines I was confused by nvm and npm, and got it wrong half the time when Aidan told me what to do...
nvm is the virtual environment manager for node/js that lets you have multiple versions active at the same time. `nvm use <version>` lets you activate the one you want.
npm is the package manager to install stuff, once you've picked the V env you want.

Reply:
@roedoejet For sure, if the latest node will do, that's what we should document and use. I'll test it that way and update my doc if it works.

Reply:
OK, tested, I am taking out the specific version and the `nvm use` statement.